posts/2020-05-27-how-to-analyze-open-liberty-logs-with-splunk.adoc

```diff
@@ -0,0 +1,139 @@
+----
+
+Breakdown of the above command::
+
+* *--log-driver=splunk* : Configures the Open Liberty container to use the Splunk log driver.
+* *--log-opt splunk-url* : The host of your Splunk Enterprise Docker container, with the port and scheme of the configured HTTP Event Collector.
+* *--log-opt splunk-token* : The Splunk HTTP Event Collector token. Ensure that *<REPLACE_WITH_SPLUNK_HEC_TOKEN>* value from the above command is replaced by the generated HEC token from Step 3 from the “Configuring the HTTP Event Collector” section.
+* *--log-opt splunk-insecureskipverify=true* : Ignores the server certificate validation, as the Splunk Enterprise container instance is using a default self-signed certificate.
+* *--log-opt splunk-format* : Specifies the message format that will get sent to Splunk’s HTTP Event Collector.
+* *WLP_LOGGING_CONSOLE_FORMAT* : An environment variable that specifies the format for the Open Liberty runtime console.
+* *WLP_LOGGING_CONSOLE_LOGLEVEL* : An environment variable which is a filter that controls the granularity of messages that go to the console.
+* *WLP_LOGGING_CONSOLE_SOURCE* : An environment variable which consists of a comma-separated list of sources that route to the console.
+
+=== Analyzing the Open Liberty container logs in Splunk
+
+In the Splunk Web Interface, navigate to *App -> Search & Reporting*.  Find the log entries aggregated from the running Open Liberty runtime container by searching with the following query, which searches for all the log entries that are from the HTTP Event Collector named `liberty` with the `line.` prefix removed from the interesting fields. The log events can be further filtered with the desired timeframe and the log information fields, as needed.
+[source, align="center"]
+----
+   source="http:liberty"  | rename line.* as *
+----
+
+image::/img/blog/blog_splunk_search.png[Splunk Search, align="center"]
+
+=== Visualizing the Open Liberty container logs in Splunk
+
+The aggregated log events from the Open Liberty Docker container can also be visualized in a collection of visualizations, which provides real-time information at a glance. I created a sample Open Liberty Splunk dashboard that can be downloaded from GitHub. This dashboard can be used with the logging data collected from JSON logging in Open Liberty. This sample dashboard visualizes message, trace, and FFDC information.
+
+. Download the sample Open Liberty Splunk dashboard from the following GitHub repository:
+https://github.com/WASdev/sample.dashboards/tree/master/Splunk%208
+
+. In Search & Reporting page, from the Splunk Web Interface, click on the *Dashboards* tab, and click *Create New Dashboard*.  In the new window that pops up, enter a title for your new dashboard and click *Create Dashboard* with the remaining fields left as-is.
```

Comment:
**change**
In the new window that pops up, enter a title for your new dashboard and click *Create Dashboard* with the remaining fields left as-is.
**to**
In the window that pops up, enter a title for your dashboard and click *Create Dashboard*, leaving the remaining fields unchanged.